Runtime/Scripts/RGBotConfigs/RGAction_KeyPress.cs

```diff
@@ -0,0 +1,298 @@
+        public bool renderDebugGizmos = false;
+
+        public Vector3 debugGizmosOffset = new Vector3(0f,-1f,2f);
+        
+#if ENABLE_INPUT_SYSTEM
+        // Allow this on any RGEntity that has an InputActionAsset
+        public InputActionAsset InputAction;
+#endif   
+        
+        private ConcurrentQueue<RG_KeyPress_Data> _keysToPress = new();
+
+        private Dictionary<Key, InputControl> _inputActions = new();
+        private InputControl _anyKey = null;
+        
+        private Dictionary<Key, double> keysDown = new();
+
+        private RGGizmos RgGizmos;
+
+        private void Start()
+        {
+#if ENABLE_INPUT_SYSTEM
+            this.RgGizmos = new();
+            if (InputAction == null)
+            {
+                throw new Exception("RGAction_KeyPress requires an InputActionAsset to be specified");
+            }
+
+            foreach (var inputActionActionMap in InputAction.actionMaps)
+            {
+                foreach (var inputAction in inputActionActionMap)
+                {
+                    foreach (var inputActionControl in inputAction.controls)
+                    {
+                        if (inputActionControl is KeyControl iac)
+                        {
+                            _inputActions.Add(iac.keyCode, inputActionControl);    
+                        }
+                        else if (inputActionControl is AnyKeyControl)
+                        {
+                            _anyKey = inputActionControl;
+                        }
+                    }
+                }
+            }
+#endif
+        }
+
+        public void Update()
+        {
+#if ENABLE_INPUT_SYSTEM
+            // one button click per frame update
+            if (_keysToPress.TryDequeue(out RG_KeyPress_Data keyToPress))
+            {
+                List<InputControl> actionsToTake = new();
+                // check for any key 
+                if (_anyKey != null)
+                {
+                    actionsToTake.Add(_anyKey);
+                }
+
+                var theKey = _inputActions.GetValueOrDefault(keyToPress.keyId, null);
+                if (theKey != null)
+                {
+                    actionsToTake.Add(theKey);
+                }
+
+                foreach (var control in actionsToTake)
+                {
+                    PressKey(keyToPress.keyId, control, keyToPress.holdTime);
+                }
+            }
+            
+            // handle un-pressing any keys that have expired their time
+            var keys = keysDown.Keys.ToList();
+            foreach (var key in keys)
+            {
+                // this is < instead of <= so that keys last at least 1 frame
+                if (keysDown[key] < Time.unscaledTime)
+                {
+                    // dictionary will have this value because we clicked it
+                    UnPressKey(key, _inputActions.GetValueOrDefault(key));
```

Comment:
That is the intent yes, which is why I keep updating the end time for that key to be release in the tracking map.  In PressKey, it doesn't send the new key down event to the input system if the key is already down... and the end time to send key up is updated accordingly.

I do apparently have a bug/deficiency in the inverse of your case though.. i'll need to fix this

What if I have the following:
t = 0, start holding "A" for 1 second
t = 0.5, start holding "A" for 0.2 second
 --- "A" currently gets released at 0.7 instead of 1.0 with the current code.
 
 ```csharp
 // set or update the un-press time tracker
var timeToUnPress = Time.unscaledTime + (holdTime > 0f ? holdTime : 0f);
// TODO: NEED CODE HERE TO VALIDATE THAT NEW VALUE IS AFTER PREVIOUS VALUE
keysDown[key] = timeToUnPress;
 ```

Reply:
@vontell ... Realized I never pushed this... now pushed in this commit
https://github.com/Regression-Games/RGUnityBots/pull/104/commits/bf9ec863a9b241ac21001e99dccf8770cb800359